cmd/juju/application/scaleapplication.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package application
+
+import (
+	"strconv"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/api/application"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/cmd/modelcmd"
+)
+
+// NewScaleApplicationCommand returns a command which scales an application's units.
+func NewScaleApplicationCommand() modelcmd.ModelCommand {
+	cmd := &scaleApplicationCommand{}
+	cmd.newAPIFunc = func() (scaleApplicationAPI, error) {
+		root, err := cmd.NewAPIRoot()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		return application.NewClient(root), nil
+	}
+	return modelcmd.Wrap(cmd)
+}
+
+// scaleApplicationCommand is responsible for destroying application units.
+type scaleApplicationCommand struct {
+	modelcmd.ModelCommandBase
+	modelcmd.CAASOnlyCommand
+
+	newAPIFunc      func() (scaleApplicationAPI, error)
+	applicationName string
+	scale           int
+}
+
+const scaleApplicationDoc = `
+Scale a Kubernetes application by specifying how many units there should be.
+
+Examples:
+
+    juju scale-application mariadb 2
+`
+
+// Info implements cmd.Command.
+func (c *scaleApplicationCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "scale-application",
+		Args:    "<application> <scale>",
+		Purpose: "Set the desired number of application units.",
+		Doc:     scaleApplicationDoc,
+	}
+}
+
+func (c *scaleApplicationCommand) Init(args []string) error {
+	if len(args) == 0 {
+		return errors.Errorf("no application specified")
+	}
+	c.applicationName = args[0]
+	if !names.IsValidApplication(c.applicationName) {
+		return errors.Errorf("invalid application name %q", c.applicationName)
+	}
+	if len(args) == 1 {
+		return errors.Errorf("no scale specified")
+	}
+	var err error
+	c.scale, err = strconv.Atoi(args[1])
+	if err != nil {
+		return errors.Annotatef(err, "invalid scale %q", args[1])
```

Comment:
Worth check for positive integer here? I am guessing negative numbers will surprise us...